packages/appsync/src/createAppSyncHooks.tsx

```diff
@@ -0,0 +1,198 @@
+import React, { useEffect, useState } from "react";
+import { createAuthLink } from "aws-appsync-auth-link";
+import { createSubscriptionHandshakeLink } from "aws-appsync-subscription-link";
+import {
+  gql,
+  ApolloClient,
+  InMemoryCache,
+  HttpLink,
+  ApolloLink,
+  useQuery as useQueryApollo,
+  useMutation as useMutationApollo,
+  useSubscription as useSubscriptionApollo,
+  QueryHookOptions,
+  MutationHookOptions,
+  ApolloProvider,
+  useApolloClient,
+  SubscriptionHookOptions,
+} from "@apollo/client";
+import { Files } from "./types";
+import { resolveFiles } from "./storageUtils";
+import { Auth } from "aws-amplify";
+
+export interface GraphqlProviderProps {
+  token?: string;
+}
+
+export const createAppSyncHooks = <
+  QT extends Record<string, { variables?: any; data: any }>,
+  MT extends Record<string, { variables?: any; data: any }>,
+  ST extends Record<string, { variables?: any; data: any }>
+>({
+  queries,
+  mutations,
+  subscriptions,
+  url,
+  region,
+  type,
+  refetchSubscriptions,
+}: {
+  queries: Record<keyof QT, string>;
+  mutations: Record<keyof MT, string>;
+  subscriptions: Record<keyof ST, string>;
+  url: string;
+  type: "AMAZON_COGNITO_USER_POOLS";
```

Comment:
did not bother for other auth provider for now but can be extended in the future if we need other providers.